common/src/main/java/indexingTopology/common/data/DataSchema.java

```diff
@@ -222,4 +251,78 @@ public boolean equals(Object object) {
 
         return true;
     }
+
+
+    public DataTuple parseJsonTuple(String tuple, String split) {
+        String[] attributes = tuple.split(split);
+        DataTuple dataTuple = new DataTuple();
+        StringBuffer stringBuffer = new StringBuffer();
+        SimpleDateFormat jsonfmt = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
+        Object attribute;
+        for (int i = 0; i < attributes.length; i++) {
+            Date date = null;
+            try {
+                date = jsonfmt.parse(attributes[i]);
+                Long lg = date.getTime();
+                attribute = dataTypes.get(i).readFromString(String.valueOf(lg));
+            } catch (ParseException e) {
+                attribute = dataTypes.get(i).readFromString(attributes[i]);
+            }
+
+            dataTuple.add(attribute);
+        }
+
+        //System.out.println(dataTuple);
+        return dataTuple;
+    }
+
+    public List<DataTuple> getTuplesFromJson(JsonObject object, String arrayName) {
+        JsonArray array = object.getAsJsonArray(arrayName);
+        List<DataTuple> dataTuples = new ArrayList<>();
+        for (JsonElement element : array) {
+            JsonObject result = element.getAsJsonObject();
+            DataTuple dataTuple = getTupleFromJson(result);
+            dataTuples.add(dataTuple);
+        }
+        return dataTuples;
+    }
+
+    public DataTuple getTupleFromJson(JsonObject object) {
+        String str = "";
+        int len = getNumberOfFields();
+        for (int i = 0; i < len; i++) {
+            String objectStr = "";
+            if(!object.get(getFieldName(i)).isJsonNull()) {
+                objectStr = object.get(getFieldName(i)).toString().replace("\"", "");
+            }
+            else {
+                objectStr = "null";
+            }
+            if(i < len - 1) {
+                str = str + objectStr + ",";
+            }
+            else {
+                str = str + objectStr;
+            }
+        }
+        DataTuple dataTuple = parseJsonTuple(str, ",");
+        return dataTuple;
+    }
+
+    public JSONObject getJsonFromDataTuple(DataTuple tuple) {
+        String jsonStr = "{";
+        int len = getNumberOfFields();
+        for (int i = 0; i < len; i++) {
```

Comment:
If my memory serve me right, JsonObject has a function to insert key value pairs. In our case, attribute name is the key, and the actual value of the attribute is the value. Use this function to construct JsonObject, instead of doing it by yourself.